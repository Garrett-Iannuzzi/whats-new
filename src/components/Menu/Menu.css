aside {
  border-top: solid 1px darkgrey;
  border-bottom: solid 1px darkgrey;
  display: flex;
  height: 3.8em;
  justify-content: space-between;
}

.news-type-btn {
  height: 2.5em;
  font-weight: bold;
  margin: 20px 10px;
  width: 15%;
}

.news-type-btn:hover {
  cursor: pointer;
  color: white;
  background-color: indianred;
}

h1 {
  color: black;
  font-size: 3em;
  padding: 15px 15px;
  text-align: left;
}

span {
  color: indianred;
}
