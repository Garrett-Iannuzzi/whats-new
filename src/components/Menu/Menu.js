import React from 'react';
import './Menu.css';
import PropTypes from 'prop-types';

const Menu = ({ filterByNewsType, btnTitles }) => {

  const menuBtns = btnTitles.map(title => {
      return (
        <button className='news-type-btn' key={Math.random().toString()} onClick={filterByNewsType}>{title}</button>
      )
  })

  return (
    <nav className='navbar'> 
      <h1>What's <span>NEW</span>s</h1>
      <aside>
        {menuBtns}
      </aside>
    </nav>
  )
}

Menu.propTypes = {
  filterByNewsType: PropTypes.func,
  btnTitles: PropTypes.array
}

export default Menu;