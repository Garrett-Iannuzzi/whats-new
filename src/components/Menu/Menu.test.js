import React from 'react';
import { shallow } from 'enzyme';
import Menu from './Menu';


describe('Menu', () => {

  let mockFilterByNewsType;
  let mockBtnTitles;
  let wrapper;

  beforeEach(() => {
    mockFilterByNewsType = jest.fn();
    mockBtnTitles = ['local news', 'entertainment', 'tech', 'science', 'health'];
    wrapper = shallow(<Menu
      filterByNewsType={mockFilterByNewsType}
      btnTitles={mockBtnTitles}
    />);
  });

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('Search Buttons', () => {
    it('should filter by correct news type when filterByNewsType is clicked, first btn', () => {
      wrapper.find('button').at(0).simulate('click');
      expect(mockFilterByNewsType).toHaveBeenCalled();
    });
  
    it('should filter by correct news type when filterByNewsType is clicked, second btn', () => {
      wrapper.find('button').at(1).simulate('click');
      expect(mockFilterByNewsType).toHaveBeenCalled();
    });
  
    it('should filter by correct news type when filterByNewsType is clicked, third btn', () => {
      wrapper.find('button').at(2).simulate('click');
      expect(mockFilterByNewsType).toHaveBeenCalled();
    });
  
    it('should filter by correct news type when filterByNewsType is clicked, fourth btn', () => {
      wrapper.find('button').at(3).simulate('click');
      expect(mockFilterByNewsType).toHaveBeenCalled();
    });
  
    it('should filter by correct news type when filterByNewsType is clicked, fifth btn', () => {
      wrapper.find('button').at(4).simulate('click');
      expect(mockFilterByNewsType).toHaveBeenCalled();
    });
  })


})