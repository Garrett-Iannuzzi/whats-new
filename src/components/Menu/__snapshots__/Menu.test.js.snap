// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should match the snapshot with all data passed in correctly 1`] = `
<nav
  className="navbar"
>
  <h1>
    What's 
    <span>
      NEW
    </span>
    s
  </h1>
  <aside>
    <button
      className="news-type-btn"
      key="0.5767024091255699"
      onClick={[MockFunction]}
    >
      local news
    </button>
    <button
      className="news-type-btn"
      key="0.41742933737744736"
      onClick={[MockFunction]}
    >
      entertainment
    </button>
    <button
      className="news-type-btn"
      key="0.9057848596123272"
      onClick={[MockFunction]}
    >
      tech
    </button>
    <button
      className="news-type-btn"
      key="0.6002011395097255"
      onClick={[MockFunction]}
    >
      science
    </button>
    <button
      className="news-type-btn"
      key="0.1700988215396515"
      onClick={[MockFunction]}
    >
      health
    </button>
  </aside>
</nav>
`;
